FantasyUniverse/Services/Authentication/FirebaseAuthService.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  FirebaseAuthService.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 17.05.2022.
+//
+
+import Foundation
+import FirebaseAuth
+import Firebase
+
+class FirebaseAuthService {
```

Comment:
Maybe consider a reactive interface using Combine or trying out the Async/Await? It would incentivise you to handle errors more gracefully - it's best to have some errors returned and adequate errors presented to the user, so I wouldn't hide them all under a boolean.